app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/vm.rb

```diff
@@ -42,6 +42,7 @@ class ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::Vm < Man
   supports :resize do
     unsupported_reason_add(:resize, _('The VM is not powered off')) unless current_state == "off"
     unsupported_reason_add(:resize, _('The VM is not connected to a provider')) unless ext_management_system
+    unsupported_reason_add(:resize, _('SAP VM resize not supported')) unless flavor.type != 'ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::SAPProfile'
```

Comment:
Prefer to use `kind_of?` rather than an exact `.type !=` check, mainly because it will still work if we subclass `ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::SAPProfile` with another more specific type of SAPProfile, or if we subclass the whole provider.  Also prefer the positive `if ==` instead of `unless !=`

```suggestion
    unsupported_reason_add(:resize, _('SAP VM resize not supported')) if flavor.kind_of?(ManageIQ::Providers::IbmCloud::PowerVirtualServers::CloudManager::SAPProfile)
```